modules/crud/sql/update.go

```diff
@@ -1,60 +1,226 @@
+			}
+			for _, val := range mm {
+				val, ok := val.(string)
+				if !ok {
+					return "", nil, utils.ErrInvalidParams
+				}
+				switch val {
+				case "date", "timestamp":
+					m[k] = val
+				default:
+					return "", nil, utils.ErrInvalidParams
+				}
+			}
+		}
+	}
+	
+	record, err := generateRecord(req.Update[op])
 	if err != nil {
 		return "", nil, err
 	}
 
 	// Generate SQL string and arguments
-	sqlString, args, err := query.ToUpdateSql(record)
+	sqlString, args, err := query.Update().Set(record).ToSQL()
 	if err != nil {
 		return "", nil, err
 	}
 
 	sqlString = strings.Replace(sqlString, "\"", "", -1)
+	switch op {
+	case "$set":
+	case "$inc":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"="+k+"+"+val, -1)
+		}
+	case "$mul":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"="+k+"*"+val, -1)
+		}
+	case "$max":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"=GREATEST("+k+","+val+")", -1)
+		}
+	case "$min":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"=LEAST("+k+","+val+")", -1)
+		}
+	case "$currentDate":
+		for k, v := range m {
+			val, ok := v.(string)
+			if !ok {
+				return "", nil, utils.ErrInvalidParams
+			}
+			if val == "timestamp" {
+				sqlString = strings.Replace(sqlString, k+"='"+val+"'", k+"=CURRENT_TIMESTAMP()", -1)
+			} else {
+				sqlString = strings.Replace(sqlString, k+"='"+val+"'", k+"=CURRENT_DATE()", -1)
+			}
+		}
+	default:
+		return "", nil, utils.ErrInvalidParams
+	}
 	return sqlString, args, nil
 }
+
+// helper function to upsert data
+func (s *SQL) upsert(ctx context.Context, project, col string, req *model.UpdateRequest, op string) error {
+	tx, err := s.client.BeginTxx(ctx, nil) //TODO - Write *sqlx.TxOption instead of nil
+	if err != nil {
+		fmt.Println("Error in initiating Transaction for Upsert")
+		return err
+	}
+
+	sqlString, args, err := s.generateUpdateQuery(ctx, project, col, req, op)
+	if err != nil {
+		return err
+	}
+	
+	stmt, err := tx.PreparexContext(ctx, sqlString)
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	result, err := stmt.ExecContext(ctx, args...)
+	if err != nil {
+		return err
+	}
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if rowsAffected == 0 {
+		if op == "$currentDate" {
+			return utils.ErrInvalidParams
+		}
+		sqlQuery, args, err := s.generateCreateQuery(ctx, project, col, &model.CreateRequest{Document: req.Update[op], Operation: utils.One})
+		if err != nil {
+			return err
+		}
+		err = doExecContext(ctx, sqlQuery, args, tx)
```

Comment:
You are firing one create query for each update op. Seems to be wrong and it won't have goot performance